### PR TITLE
Combine into VPUSH for s16 and s32 as well

### PR DESCRIPTION
**Pull request by abhinay-anubola**
Add support for 16-bit and 32-bit VInserts in PreLegalizerCombiner

### CODE REVIEW

File: llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp [hunk 1]
```diff
@@ -202,7 +206,7 @@ AIE2PreLegalizerCombinerImpl::getVectorInsertIndices(
                                        Intrinsic::aie2_set_I512_I256);
   };
 
-  while (Is8BitVInsert(CurMI)) {
+  while (IsVInsert(CurMI)) {
```
Comment (gbossu) [inline llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp]: Does this mean we are allowing a mix in VINSERT instructions?

File: llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp [hunk 2]
```diff
@@ -286,6 +300,12 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineIntrinsic(
   case Intrinsic::aie2_vinsert8_I512: {
```
Comment (andcarminati) [inline llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp]: I was wondering if we can refactor in some way this scalar size information. We use this information here and in `IsVInsert`. If we have something like `getVInsertScalarSize(...)` that returns zero if the intrinsic is not know would help. For example, in your previous lambda will be enough use `return getVInsertScalarSize(cast<GIntrinsic>(*MI).getIntrinsicID()) == SclSrcBits;` You can also drop your switch-case here and pass directly the value of `getVInsertScalarSize(...)` to `tryToCombineVectorInserts` if it is greater than 0.

**Comment by andcarminati** [comment]
This PR looks good! Do you have some QoR numbers?

**Comment by abhinay-anubola** [comment]
There is no change in QoR for now.

File: llvm/test/CodeGen/AIE/aie2/GlobalISel/combine-vinsert-sequence-prelegalizer.mir [hunk 3]
```diff
@@ -328,3 +328,375 @@ body:             |
...
+    %28:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %27(<16 x s32>), %12(s32), %7(s32)
+    %29:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %28(<16 x s32>), %13(s32), %9(s32)
+    %30:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %29(<16 x s32>), %14(s32), %11(s32)
+    %31:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %30(<16 x s32>), %15(s32), %13(s32)
+    %32:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %31(<16 x s32>), %16(s32), %15(s32)
+    PseudoRET implicit $lr, implicit %32
+...
+
+---
+name:            vinsert16-I128_idx_miss
```
Comment (martien-de-jong) [inline llvm/test/CodeGen/AIE/aie2/GlobalISel/combine-vinsert-sequence-prelegalizer.mir:440]: Perhaps a very short description of why these cases do not yield a nice G_BUILD_VECTOR_TRUNC

**Review by martien-de-jong — DISMISSED** [review]
Just one nit

**Review by martien-de-jong — APPROVED** [review]

**Pull request merged**